src/olympia/abuse/models.py

```diff
@@ -235,6 +231,20 @@ def notify_reporters(self, action_helper):
             is_appeal=True,
         )
 
+    def handle_job_recreated(self, new_job_id):
+        new_job, _ = CinderJob.objects.update_or_create(
+            job_id=new_job_id,
+            defaults={
+                'resolvable_in_reviewer_tools': True,
+                'target_addon': self.target_addon,
+            },
+        )
+        # Update our fks to connected objects
+        AbuseReport.objects.filter(cinder_job=self).update(cinder_job=new_job)
```

Comment:
These fk updates feel a bit hacky... but the alternative is everywhere we check for reports attached to a job we have to also navigate down a possible `forwarded_to_job` fk - similarly for CinderDecision.appeal_job.  That seemed worse to me.